File: 프로그래머스/2/147354. 테이블 해시 함수/테이블 해시 함수.py
def solution(data, col, row_begin, row_end):
    data.sort(key = lambda x : (x[col-1], -x[0])) # 조건에 따라 정렬
    
    answer = 0
    for i in (data[row_begin-1]):
        answer += i % row_begin    
        
    for i in range(row_begin, row_end):
        tmp = 0
        for j in (data[i]):
            tmp += j % (i+1)
        answer = answer^tmp
        
    return answer

File: 프로그래머스/2/147354. 테이블 해시 함수/테이블 해시 함수.java
import java.util.Arrays;

class Solution {
    public int solution(int[][] data, int col, int row_begin, int row_end) {
        int answer = 0;
        Arrays.sort(data, (a, b) -> a[col-1] != b[col-1] ? a[col-1] - b[col-1] : b[0] - a[0]); // 정렬
        for (int i = row_begin-1; i < row_end; i++) {
            int[] S = data[i];
            int Si = 0;
            for (int j : S) {
                Si += j % (i+1);
            }
            answer = answer ^ Si; 
        }
        
        
        
        return answer;
    }
}